### PR TITLE
Make autotools check for all the used openssl functions

### LINKED ISSUE #836: tests.c: undefined reference to `ECSDA_SIG_get0`
```
Build Options:
  with ecmult precomp     = yes
  with external callbacks = no
  with benchmarks         = yes
  with coverage           = no
  module ecdh             = no
  module recovery         = no
  module extrakeys        = no
  module schnorrsig       = no

  asm                     = x86_64
  bignum                  = gmp
  ecmult window size      = 15
  ecmult gen prec. bits   = 4

  valgrind                = no
  CC                      = gcc
  CFLAGS                  = -O2 -fvisibility=hidden -std=c89 -pedantic -Wall -Wextra -Wcast-align -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef -Wno-unused-function -Wno-long-long -W
no-overlength-strings -W -g
  CPPFLAGS                = 
  LDFLAGS                 = 
```

```
  CCLD     tests
src/tests-tests.o: In function `test_ecdsa_der_parse':
src/tests.c:4750: undefined reference to `ECDSA_SIG_get0'
src/tests.c:4751: undefined reference to `BN_is_negative'
src/tests.c:4751: undefined reference to `BN_is_negative'
collect2: error: ld returned 1 exit status
```

```
$ git rev-parse HEAD
c6b6b8f1bb044d7d1aa065ebb674adde98a36a8e
$ cat /etc/redhat-release 
Red Hat Enterprise Linux Workstation release 7.7 (Maipo)
```

### PR DESCRIPTION
**Pull request by elichai**
I added all the openssl functions that we call in `tests.c` and in `bench_verify.c` to the m4 check, that way if any of them are missing it won't enable openssl.
I also modified it a little to prevent a segmentation fault when running that program (not that it really matters for autotools)

This should fix #836

### CODE REVIEW

**Review by real-or-random — CHANGES_REQUESTED** [review]
`d2i_ECPrivateKey` and `EC_KEY_check_key` (used in `get_openssl_key` in tests.c) are missing.

**Comment by elichai** [comment]
> `d2i_ECPrivateKey` and `EC_KEY_check_key` (used in `get_openssl_key` in tests.c) are missing.

I have no idea how I missed this, I grepped `ENABLE_OPENSSL_TESTS` multiple times.
Will add now, thanks!

**Review by real-or-random — APPROVED** [review]
ACK e6692778d3f6507eb1325785cdd424073a945ff7 

I've just realized that the summary of build options at the end of configure does not show whether tests are enabled, nor whether OpenSSLs are enabled. If you want, add a commit.

**Comment by jonasnick** [comment]
> If you want, add a commit.

Do you @elichai ? :) I think that the the author of #836 might have noticed that this was a configure option if it was diplayed in the summary.

**Comment by elichai** [comment]
Is this what you meant? @real-or-random @jonasnick 

It's also somewhat confusing that everyone calls the library `openssl` but we actually use libcrypto (which is a part of openssl), so the logs say:
```
checking for main in -lcrypto... yes                                                                                  
checking for EC functions in libcrypto... yes                                                                         
```

File: configure.ac [hunk 1]
```diff
@@ -397,6 +397,7 @@ if test x"$use_tests" = x"yes"; then
   SECP_OPENSSL_CHECK
   if test x"$has_openssl_ec" = x"yes"; then
     if test x"$enable_openssl_tests" != x"no"; then
+    enable_openssl_tests=yes
```
Comment (jonasnick) [inline configure.ac:400]: wrong indentation

**Review by real-or-random — APPROVED** [review]
ACK 3734b68200ee37f5eea80f47d611e9b5a65548fe

**Comment by sipa** [comment]
ACK 3734b68200ee37f5eea80f47d611e9b5a65548fe

**Pull request merged**